CMakeLists.txt

```diff
@@ -16,23 +16,48 @@ option(ENABLE_NESO_TESTS
 include(cmake/Sanitizers.cmake)
 include(cmake/CheckFileList.cmake)
 
+
+
 # ##############################################################################
 # Find dependencies
 # ##############################################################################
-find_package(SYCL REQUIRED)
+
+#USE -DNESO_SYCL_VENDOR to chose implementation at gen time
+set(NESO_SYCL_VENDOR "Default" CACHE STRING "SYCL vendor to look for - required
+if present")
+include(cmake/SYCL.cmake)
+if (NESO_SYCL_VENDOR MATCHES "Default")
+    find_sycl()
+else()
+    find_sycl(${NESO_SYCL_VENDOR})
+endif()
 if(SYCL_FOUND)
-  message(STATUS "Using SYCL implementation ${SYCL_IMPLEMENTATION}")
-  if(hipSYCL_FOUND)
-    set(HIPSYCL_TARGETS "omp")
-    set(SYCL_FLAG NESO_HIPSYCL)
-  elseif(SYCL_IMPLEMENTATION STREQUAL "IntelDPCPP")
-    set(SYCL_FLAG NESO_DPCPP)
-  endif()
+    message(STATUS "Using SYCL implementation ${SYCL_IMPLEMENTATION}")
+    if(NOT SYCL_INTEL)
+        set(HIPSYCL_TARGETS "omp")
+        set(ACPP_TARGETS "omp")
+        set(SYCL_FLAG NESO_HIPSYCL)
+    else()
+        set(SYCL_FLAG NESO_DPCPP)
+    endif()
```

Comment:
We've spoken about this briefly in person, but I'm not clear on the logic of taking this out of the FindSYCL file.

Reply:
The more I think about it the more I dislike the idea of FindSYCL *and* the find_sycl macro I wrote - I changed it so I could explicitly ask for a particular SYCL version at "cmake time" and I couldn't think of a standard way to do that with a find_package - its not really a component or a version. 

I think we should just force people to chose - SYCL isn't standardised enough for a find mpi style construct to work well IMO. As I mentioned in another comment it's too easy to get something unexpected this way - setting targets for hipsycl but finding intelSYCL. 

Having said that - what I have here is broken in a couple of different ways so I will fix it.